js/src/files/add.js

```diff
@@ -120,6 +120,26 @@ module.exports = (createCommon, options) => {
       })
     })
 
+    it('should not be able to add a string', (done) => {
+      const data = `TEST${Date.now()}`
+
+      ipfs.files.add(data, (err) => {
+        expect(err).to.exist()
+        expect(err.message).to.contain('invalid input')
+        done()
+      })
+    })
+
+    it('should not be able to add a non-Buffer TypedArray', (done) => {
```

Comment:
Should this test include more of the different typed arrays?

![image](https://user-images.githubusercontent.com/459764/47350165-8b579e80-d6b5-11e8-8d77-6ebd9f63a319.png)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Typed_arrays#Typed_array_views

Reply:
I think that since these all inherit from `TypedArray` then testing with just one is good enough, the point here is to ensure the API is feeding back with an appropriate message and not passing the input further down the stack where it'll cause an error that doesn't communicate the real problem.